coalib/settings/Section.py

```diff
@@ -159,6 +160,10 @@ def add_or_create_setting(self,
             val.value = str(val._value) + '\n' + setting._value
             self.append(val, custom_key=key)
         else:
+            if key in list(self.contents.keys()):
```

Comment:
`allow_appending` will not always be `True`. It depends on whether `keys` is an empty list or not. See https://github.com/coala/coala/blob/master/coalib/parsing/ConfParser.py#L137. It will be an empty list if the setting in a section spans multiple lines. The `.coafile` is parsed line by line, `allow_appending` will only be `True` if the setting spans multiple lines and all the lines need to be mapped to a single key.